flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSplitPlanner.java

```diff
@@ -86,10 +86,18 @@ static CloseableIterable<CombinedScanTask> planTasks(
       IncrementalAppendScan scan = table.newIncrementalAppendScan();
       scan = refineScanWithBaseConfigs(scan, context, workerPool);
 
+      if (context.startTag() != null && table.snapshot(context.startTag()) != null) {
```

Comment:
I think we probably need to verify that user only sets one of `START_SNAPSHOT_ID` and `START_TAG`, instead of both. Also the same for end case.